dspace-api/src/main/java/org/dspace/health/LicenseCheck.java

```diff
@@ -38,92 +37,85 @@ public class LicenseCheck extends Check {
+            for (UUID uuid : bitstreamUUIDs) {
+                ClarinLicenseResourceMapping mapping = mappingByUUID.get(uuid);
+                if (mapping == null) {
                     problemItems.computeIfAbsent(
-                            "UUIDs of bitstreams without license mappings", k -> new ArrayList<>()).add(uuid);
+                        "UUIDs of bitstreams without license mappings", k -> new ArrayList<>()).add(uuid);
                     continue;
                 }
-
                 // Every resource mapping between license and the bitstream has only one record,
                 // because the bitstream has unique UUID, so get the first record from the List
-                ClarinLicenseResourceMapping clarinLicenseResourceMapping = clarinLicenseResourceMappingList.get(0);
-
-                ClarinLicenseLabel nonExtendedLabel =
-                        clarinLicenseResourceMapping.getLicense().getNonExtendedClarinLicenseLabel();
-
+                ClarinLicenseLabel nonExtendedLabel = mapping.getLicense().getNonExtendedClarinLicenseLabel();
                 if (Objects.isNull(nonExtendedLabel)) {
-                    log.error("Item {} with id {} does not have non extended license label.",
-                            item.getName(), item.getID());
+                    problemItems.computeIfAbsent(
+                            "UUIDs of bitstreams without non-extended license labels",
+                            k -> new ArrayList<>()).add(uuid);
                 } else {
                     licensesCount.put(nonExtendedLabel.getLabel(),
                             licensesCount.getOrDefault(nonExtendedLabel.getLabel(), 0) + 1);
                 }
-            } catch (SQLException e) {
-                throw new RuntimeException("Error while fetching ClarinLicenseResourceMapping by Bitstream UUID: " +
-                        uuid, e);
             }
+        } catch (SQLException e) {
+            throw new RuntimeException("Error while fetching items or license mappings.", e);
+        } finally {
+            context.close();
         }
-
         for (Map.Entry<String, Integer> result : licensesCount.entrySet()) {
             sb.append(String.format("%-20s: %d\n", result.getKey(), result.getValue()));
         }
-
         if (!problemItems.isEmpty()) {
-            for (Map.Entry<String, List<UUID>> problemItems : problemItems.entrySet()) {
-                List<UUID> uuids = problemItems.getValue();
-                sb.append(String.format("\n%s: %d\n", problemItems.getKey(), uuids.size()));
+            for (Map.Entry<String, List<UUID>> problemEntry : problemItems.entrySet()) {
+                List<UUID> uuids = problemEntry.getValue();
+                sb.append(String.format("\n%s: %d\n", problemEntry.getKey(), uuids.size()));
                 for (UUID uuid : uuids) {
                     sb.append(String.format("     %s\n", uuid));
                 }
             }
         }
-
-        context.close();
```

Comment:
Why did you remove it?

Reply:
No it is there but in finally block: https://github.com/dataquest-dev/DSpace/pull/1135/files/2ebe1c05f4acf461b1479a03d4ef15e27f192936#diff-73560a06b89a19a9ca979b6810f8f62d3d8456a30289138b8d11520a7a95b7c6R105